qwerty_ast_to_mlir/src/lower/qpu.rs

```diff
@@ -434,44 +428,90 @@ fn is_bell_basis(vecs: &[Vector]) -> bool {
     }
 }
 
-// TODO: Remove this hack
-/// Returns `true` if this is the Fourier basis as defined in the QCE '25
-/// prelude using the `revolve` basis generator.
-fn is_fourier(basis: &Basis) -> bool {
-    match basis {
-        // {'0'+'1', '0'-'1'} == pm (the fourier[1] base case)
-        Basis::BasisLiteral { vecs, .. } if vecs.len() == 2 => {
-            (if let Vector::UniformVectorSuperpos { q1, q2, .. } = &vecs[0] {
-                matches!(&**q1, Vector::ZeroVector { .. })
-                    && matches!(&**q2, Vector::OneVector { .. })
-            } else {
-                false
-            }) && (if let Vector::UniformVectorSuperpos { q1, q2, .. } = &vecs[1] {
-                matches!(&**q1, Vector::ZeroVector { .. })
-                    && if let Vector::VectorTilt { q, angle_deg, .. } = &**q2 {
-                        matches!(&**q, Vector::OneVector { .. })
-                            && angles_are_approx_equal(*angle_deg, 180.0)
-                    } else {
-                        false
-                    }
-            } else {
-                false
-            })
+/// If the Vec of Bases is a vector of only basis literals {0, 1},
+/// or only {p, m}, or only {i, j}, then we want to find this!
+fn is_vec_basis_equiv_primitive(basis_elems: &Vec<Basis>) -> (bool, qwerty::PrimitiveBasis) {
```

Comment:
My brain is reading "vec" as "basis vector"
```suggestion
fn is_basis_primitive(basis_elems: &[Basis]) -> (bool, qwerty::PrimitiveBasis) {
```

Reply:
I agree with returning an option instead of a bool, but see below comment for changing it to one basis vs many.